python/cugraph/cugraph/sampling/node2vec.py

```diff
@@ -0,0 +1,140 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pylibcugraph
+import cudf
+from cugraph.utilities import ensure_cugraph_obj_for_nx
+
+
+def node2vec(G, start_vertices, max_depth, use_padding, p=1.0, q=1.0):
+    """
+    Computes random walks for each node in 'start_vertices', under the
+    node2vec sampling framework.
+
+    References
+    ----------
+
+    A Grover, J Leskovec: node2vec: Scalable Feature Learning for Networks,
+    Proceedings of the 22nd ACM SIGKDD International Conference on Knowledge
+    Discovery and Data Mining, https://arxiv.org/abs/1607.00653
+
+    Parameters
+    ----------
+    G : cuGraph.Graph or networkx.Graph
+        The graph can be either directed (DiGraph) or undirected (Graph).
+
+    start_vertices: int or list or cudf.Series
+        A single node or a list or a cudf.Series of nodes from which to run
+        the random walks
+
+    max_depth: int, optional
```

Comment:
follow up with the declaration from the function declaration I commented

Reply:
I assume you mean the description in the docstring right? I think I did that because the default value is None, but as you mentioned, `max_depth` certainly isn't optional, so I'll go ahead and remove that from node2vec and random_walks